### ISSUE TITLE
Cannot establish a connection with explicit HTTPS port in URL as 443

### ISSUE THREAD

**Issue by nobuto-m** [minio/mc#4585]
## Expected behavior

`https://<MY_ENDPOINT>:443` as URL shouldn't cause any issue.

## Actual behavior

By having `https://<MY_ENDPOINT>:443` for URL, the server side returns 400 Bad Request although `https://<MY_ENDPOINT>` just works.

It's a cosmetic issue once we know the culprit, but it was hard to notice `:443` was the one.

## Steps to reproduce the behavior

1. configure the client in the following way with an explicit port 443.

```
$ mc alias list my-test
my-test
  URL       : https://192.168.151.107:443
  AccessKey : W61LAPYFM9LAX8GA6810
  SecretKey : D4TXAT5xhIPWHX9t5nNgPxS1T19EA7sddxPMa7rK
  API       : s3v4
  Path      : auto
```

2. try to list buckets.

```
$ mc --insecure --debug ls my-test
mc: <DEBUG> GET / HTTP/1.1
Host: [192.168.151.107]
User-Agent: MinIO (linux; amd64) minio-go/v7.0.55 mc/RELEASE.2023-05-26T23-31-54Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20230530/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20230530T060458Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 400 Bad Request
Connection: close
Content-Length: 308
Content-Type: text/html; charset=iso-8859-1
Date: Tue, 30 May 2023 06:04:58 GMT
Server: Apache/2.4.52 (Ubuntu)

<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>400 Bad Request</title>
</head><body>
<h1>Bad Request</h1>
<p>Your browser sent a request that this server could not understand.<br />
</p>
<hr>
<address>Apache/2.4.52 (Ubuntu) Server at 192.168.151.107 Port 443</address>
</body></html>
mc: <DEBUG> TLS Certificate found: 
mc: <DEBUG>  >> Expires: 2024-05-29 02:57:00 +0000 UTC
mc: <DEBUG> Response Time:  4.15958ms
```

By removing `:443` from the URL, it works.

```
$ mc --insecure --debug ls my-test
mc: <DEBUG> GET / HTTP/1.1
Host: 192.168.151.107
User-Agent: MinIO (linux; amd64) minio-go/v7.0.55 mc/RELEASE.2023-05-26T23-31-54Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20230530/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20230530T060725Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Transfer-Encoding: chunked
Content-Type: application/xml
Date: Tue, 30 May 2023 06:07:25 GMT
Server: Apache/2.4.52 (Ubuntu)
Vary: Accept-Encoding
X-Amz-Request-Id: tx0000016e2936c8eef7bfc-006475929d-1a70-default

mc: <DEBUG> TLS Certificate found: 
mc: <DEBUG>  >> Expires: 2024-05-29 02:57:00 +0000 UTC
mc: <DEBUG> Response Time:  6.168302ms

[2023-05-30 05:19:01 UTC]     0B my-test/
```

## mc --version
```
$ mc --version
mc version RELEASE.2023-05-26T23-31-54Z (commit-id=9cb069e7afaa45c64c45b7b59ba65c0441019efd)
Runtime: go1.19.9 linux/amd64
Copyright (c) 2015-2023 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
```

## System information
- client: Ubuntu 22.04 LTS
- S3 server: Ceph radosgw (quincy) + Apache2 as a reverse proxy.

**Comment by vadmeste** [comment]
> Host: [192.168.151.107]

I think this is the culprit, apache is complaining about this, nginx won't. I will fix this quickly.

**Comment by vadmeste** [comment]
This is fixed since minio-go was updated to v7.0.56
